File: packages/web/src/modules/service-requests/types/wizard-session.ts
/**
 * Wizard Session Types - Cache-First Architecture
 *
 * Matches backend models in:
 *   packages/backend/app/modules/service_requests/models/wizard_session.py
 *
 * Backend responses use snake_case. These types use camelCase for frontend.
 *
 * @since v2.0 - Cache-first wizard migration
 * @see .claude/plans/CACHE_FIRST_WIZARD_MIGRATION_PLAN.md
 */

// ============================================================================
// ENUMS
// ============================================================================

export enum WizardSessionStatus {
  ACTIVE = 'active',
  DOCUMENTS_UPLOADED = 'documents_uploaded',
  READY_FOR_PAYMENT = 'ready_for_payment',
  PAYMENT_INITIATED = 'payment_initiated',
  PERSISTED = 'persisted',
  EXPIRED = 'expired',
  CANCELLED = 'cancelled',
}

// ============================================================================
// BACKEND RESPONSE TYPES (snake_case - direct from API)
// ============================================================================

/** Matches WizardSessionResponse in wizard_session.py */
export interface BackendWizardSessionResponse {
  session_id: string
  workflow_code: string
  solicitud_type: string
  sub_type: string | null
  motivo: string | null
  is_minor: boolean

  status: WizardSessionStatus
  current_step: number
  current_step_id: string | null

  documents_count: number
  documents_uploaded: string[]

  form_data: Record<string, unknown>
  extracted_data: Record<string, unknown>

  tariff: TariffBreakdown | null

  created_at: string
  updated_at: string
  expires_at: string
  ttl_seconds: number

  required_documents: BackendRequiredDocument[]

  // Workflow capabilities
  requires_appointment: boolean
  entity_code: string | null
}

export interface BackendRequiredDocument {
  code: string
  name_es: string
  is_required: boolean
  uploaded: boolean
}

/** Matches WizardDocumentPreviewResponse */
export interface BackendDocumentPreviewResponse {
  session_id: string
  document_code: string
  document_name: string | null
  file_name: string
  file_size: number

  extraction: Record<string, unknown>
  confidence: number
  processor: string
  extraction_status: string
  needs_correction: boolean

  risk_analysis: Record<string, unknown> | null
  cross_validation: Record<string, unknown> | null

  expires_at: string
  ttl_seconds: number
}

/** Matches WizardPreparePaymentResponse */
export interface BackendPreparePaymentResponse {
  session_id: string
  ready_for_payment: boolean

  tariff: TariffBreakdown
  total_amount: number
  currency: string

  validation_passed: boolean
  errors: ValidationItem[]
  warnings: ValidationItem[]

  all_documents_uploaded: boolean
  missing_documents: string[]
}

/** Matches WizardPersistResult */
export interface BackendPersistResult {
  success: boolean
  service_request_id: string | null
  reference: string | null
  payment_id: string | null
  error: string | null
  error_code: string | null
}

// ============================================================================
// FRONTEND TYPES (camelCase - used in components)
// ============================================================================

export interface WizardSession {
  sessionId: string
  workflowCode: string
  solicitudType: string
  subType: string | null
  motivo: string | null
  isMinor: boolean

  status: WizardSessionStatus
  currentStep: number
  currentStepId: string | null

  documentsCount: number
  documentsUploaded: string[]

  formData: Record<string, unknown>
  extractedData: Record<string, unknown>

  tariff: TariffBreakdown | null

  createdAt: string
  updatedAt: string
  expiresAt: string
  ttlSeconds: number

  requiredDocuments: RequiredDocument[]

  // Workflow capabilities
  requiresAppointment: boolean
  entityCode: string | null
}

export interface RequiredDocument {
  code: string
  nameEs: string
  isRequired: boolean
  uploaded: boolean
}

export interface DocumentPreview {
  sessionId: string
  documentCode: string
  documentName: string | null
  fileName: string
  fileSize: number

  extraction: Record<string, unknown>
  confidence: number
  processor: string
  extractionStatus: string
  needsCorrection: boolean

  riskAnalysis: Record<string, unknown> | null
  crossValidation: Record<string, unknown> | null

  expiresAt: string
  ttlSeconds: number
}

export interface PreparePaymentResult {
  sessionId: string
  readyForPayment: boolean

  tariff: TariffBreakdown
  totalAmount: number
  currency: string

  validationPassed: boolean
  errors: ValidationItem[]
  warnings: ValidationItem[]

  allDocumentsUploaded: boolean
  missingDocuments: string[]
}

export interface PersistResult {
  success: boolean
  serviceRequestId: string | null
  reference: string | null
  paymentId: string | null
  error: string | null
  errorCode: string | null
}

// ============================================================================
// SHARED TYPES
// ============================================================================

export interface TariffBreakdown {
  base_amount: number
  supplements: TariffSupplement[]
  total_amount: number
  currency: string
  source?: string
}

export interface TariffSupplement {
  code: string
  label_es: string
  amount: number
}

export interface ValidationItem {
  rule_id: string
  message_es: string
  field: string | null
}

// ============================================================================
// REQUEST TYPES (sent to backend)
// ============================================================================

export interface WizardSessionCreateRequest {
  workflow_code: string
  solicitud_type?: string
  sub_type?: string | null
  motivo?: string | null
  is_minor?: boolean
}

export interface DocumentConfirmRequest {
  document_code: string
  confirmed_data: Record<string, unknown>
  user_notes?: string | null
}

export interface FormDataSaveRequest {
  form_data: Record<string, unknown>
  step_id?: string | null
}

// ============================================================================
// TRANSFORM FUNCTIONS (Backend snake_case → Frontend camelCase)
// ============================================================================

export function transformSession(
  backend: BackendWizardSessionResponse
): WizardSession {
  return {
    sessionId: backend.session_id,
    workflowCode: backend.workflow_code,
    solicitudType: backend.solicitud_type,
    subType: backend.sub_type,
    motivo: backend.motivo,
    isMinor: backend.is_minor,
    status: backend.status,
    currentStep: backend.current_step,
    currentStepId: backend.current_step_id,
    documentsCount: backend.documents_count,
    documentsUploaded: backend.documents_uploaded,
    formData: backend.form_data,
    extractedData: backend.extracted_data,
    tariff: backend.tariff,
    createdAt: backend.created_at,
    updatedAt: backend.updated_at,
    expiresAt: backend.expires_at,
    ttlSeconds: backend.ttl_seconds,
    requiredDocuments: (backend.required_documents || []).map(transformRequiredDocument),
    requiresAppointment: backend.requires_appointment ?? false,
    entityCode: backend.entity_code ?? null,
  }
}

function transformRequiredDocument(
  backend: BackendRequiredDocument
): RequiredDocument {
  return {
    code: backend.code,
    nameEs: backend.name_es,
    isRequired: backend.is_required,
    uploaded: backend.uploaded,
  }
}

export function transformDocumentPreview(
  backend: BackendDocumentPreviewResponse
): DocumentPreview {
  return {
    sessionId: backend.session_id,
    documentCode: backend.document_code,
    documentName: backend.document_name,
    fileName: backend.file_name,
    fileSize: backend.file_size,
    extraction: backend.extraction,
    confidence: backend.confidence,
    processor: backend.processor,
    extractionStatus: backend.extraction_status,
    needsCorrection: backend.needs_correction,
    riskAnalysis: backend.risk_analysis,
    crossValidation: backend.cross_validation,
    expiresAt: backend.expires_at,
    ttlSeconds: backend.ttl_seconds,
  }
}

export function transformPreparePayment(
  backend: BackendPreparePaymentResponse
): PreparePaymentResult {
  return {
    sessionId: backend.session_id,
    readyForPayment: backend.ready_for_payment,
    tariff: backend.tariff,
    totalAmount: backend.total_amount,
    currency: backend.currency,
    validationPassed: backend.validation_passed,
    errors: backend.errors,
    warnings: backend.warnings,
    allDocumentsUploaded: backend.all_documents_uploaded,
    missingDocuments: backend.missing_documents,
  }
}

export function transformPersistResult(
  backend: BackendPersistResult
): PersistResult {
  return {
    success: backend.success,
    serviceRequestId: backend.service_request_id,
    reference: backend.reference,
    paymentId: backend.payment_id,
    error: backend.error,
    errorCode: backend.error_code,
  }
}

// ============================================================================
// CROSS-TYPE TRANSFORMS (Session ↔ Legacy)
// ============================================================================

/**
 * Transform a session DocumentPreview to the legacy DocumentExtractionPreview
 * format used by DocumentPreviewDialog.
 *
 * This adapter allows reusing the existing DocumentPreviewDialog component
 * with the cache-first wizard's preview response.
 */
export function transformPreviewToExtractionPreview(
  preview: DocumentPreview
): {
  previewId: string
  documentCode: string
  documentName: string
  fileName: string
  fileSize: number
  mimeType: string
  extraction: Record<string, unknown>
  confidence: number
  processor: 'gemini' | 'tesseract' | 'hybrid'
  fieldIndicators: Array<{
    fieldName: string
    value?: unknown
    confidence: number
    status: 'ok' | 'warning' | 'error' | 'missing'
    riskLevel?: 'low' | 'medium' | 'high' | 'critical'
    riskMessage?: string
    requiresAttention: boolean
    suggestion?: string
  }>
  riskAnalysis?: {
    riskLevel: string
    riskScore: number
    riskFactors: Array<{ type: string; severity: string; description: string }>
    recommendations: string[]
    requiresRejection: boolean
    requiresReview: boolean
    factorsCount: Record<string, number>
    identityMismatches?: Array<{
      field_name: string
      field_label: { es: string; fr: string; en: string }
      is_blocking: boolean
      source_document: { code: string; value: string }
      compared_document: { code: string; value: string }
      risk_code: string
      severity: string
    }>
    hasBlockingMismatches?: boolean
  }
  extractionStatus: string
  needsCorrection: boolean
  detectedDocumentType?: string
  documentTypeMatch: boolean
  expectedFields: Array<{ key: string; label: string; type: string; required: boolean }>
  expiresAt: string
  processingTimeMs?: number
} {
  // Parse risk analysis from generic Record to typed structure
  const ra = preview.riskAnalysis as Record<string, unknown> | null
  const riskAnalysis = ra ? {
    riskLevel: (ra.risk_level as string) || (ra.riskLevel as string) || 'low',
    riskScore: (ra.risk_score as number) || (ra.riskScore as number) || 0,
    riskFactors: (ra.risk_factors as Array<{ type: string; severity: string; description: string }>) ||
      (ra.riskFactors as Array<{ type: string; severity: string; description: string }>) || [],
    recommendations: (ra.recommendations as string[]) || [],
    requiresRejection: (ra.requires_rejection as boolean) || (ra.requiresRejection as boolean) || false,
    requiresReview: (ra.requires_review as boolean) || (ra.requiresReview as boolean) || false,
    factorsCount: (ra.factors_count as Record<string, number>) || (ra.factorsCount as Record<string, number>) || {},
    identityMismatches: (ra.identity_mismatches as Array<{
      field_name: string
      field_label: { es: string; fr: string; en: string }
      is_blocking: boolean
      source_document: { code: string; value: string }
      compared_document: { code: string; value: string }
      risk_code: string
      severity: string
    }>) || (ra.identityMismatches as unknown as Array<{
      field_name: string
      field_label: { es: string; fr: string; en: string }
      is_blocking: boolean
      source_document: { code: string; value: string }
      compared_document: { code: string; value: string }
      risk_code: string
      severity: string
    }>) || undefined,
    hasBlockingMismatches: (ra.has_blocking_mismatches as boolean) || (ra.hasBlockingMismatches as boolean) || false,
  } : undefined

  // Build field indicators from extraction data
  const fieldIndicators = Object.entries(preview.extraction).map(([key, value]) => ({
    fieldName: key,
    value,
    confidence: preview.confidence,
    status: (preview.confidence >= 0.7 ? 'ok' : 'warning') as 'ok' | 'warning' | 'error' | 'missing',
    requiresAttention: preview.confidence < 0.7,
  }))

  return {
    previewId: `${preview.sessionId}-${preview.documentCode}`,
    documentCode: preview.documentCode,
    documentName: preview.documentName || preview.documentCode,
    fileName: preview.fileName,
    fileSize: preview.fileSize,
    mimeType: 'application/octet-stream', // Not available from session preview
    extraction: preview.extraction,
    confidence: preview.confidence,
    processor: preview.processor as 'gemini' | 'tesseract' | 'hybrid',
    fieldIndicators,
    riskAnalysis,
    extractionStatus: preview.extractionStatus,
    needsCorrection: preview.needsCorrection,
    documentTypeMatch: true, // Default - session preview doesn't have this
    expectedFields: [], // Not available from session preview
    expiresAt: preview.expiresAt,
  }
}

// ============================================================================
// UI HELPERS
// ============================================================================

export function getSessionStatusColor(status: WizardSessionStatus): string {
  switch (status) {
    case WizardSessionStatus.ACTIVE:
      return 'blue'
    case WizardSessionStatus.DOCUMENTS_UPLOADED:
      return 'cyan'
    case WizardSessionStatus.READY_FOR_PAYMENT:
      return 'green'
    case WizardSessionStatus.PAYMENT_INITIATED:
      return 'orange'
    case WizardSessionStatus.PERSISTED:
      return 'green'
    case WizardSessionStatus.EXPIRED:
      return 'red'
    case WizardSessionStatus.CANCELLED:
      return 'gray'
    default:
      return 'gray'
  }
}

File: packages/backend/app/modules/service_requests/models/wizard_session.py
"""
Pydantic models for Wizard Session (Cache-First Architecture).

These models support the cache-first wizard where all data stays in Redis
until payment is initiated. No DB/Firebase writes until user confirms payment.

@since v2.0 - Cache-first wizard migration
@see .claude/plans/CACHE_FIRST_WIZARD_MIGRATION_PLAN.md
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime, date, time
from uuid import UUID
from enum import Enum


# =============================================================================
# ENUMS
# =============================================================================

class WizardSessionStatus(str, Enum):
    """Status of a wizard session."""
    ACTIVE = "active"              # Session is active, user is working
    DOCUMENTS_UPLOADED = "documents_uploaded"  # All documents uploaded
    READY_FOR_PAYMENT = "ready_for_payment"    # Form reviewed, ready to pay
    PAYMENT_INITIATED = "payment_initiated"    # Payment started
    PERSISTED = "persisted"        # Data persisted to DB (success)
    EXPIRED = "expired"            # Session expired (TTL)
    CANCELLED = "cancelled"        # User cancelled


# =============================================================================
# SESSION DATA MODELS
# =============================================================================

class SessionDocumentData(BaseModel):
    """Document data stored in session cache."""
    document_code: str
    document_name: Optional[str] = None
    file_name: str
    file_size: int
    mime_type: str
    content_b64: str = Field(..., description="Base64-encoded file content")

    # Extraction results
    extraction: Dict[str, Any] = Field(default_factory=dict)
    confidence: float = Field(default=0, ge=0, le=1)
    processor: str = Field(default="unknown", description="gemini, tesseract, hybrid")
    extraction_status: str = Field(default="pending", description="pending, success, failed, manual_review")

    # Risk analysis
    risk_analysis: Optional[Dict[str, Any]] = None

    # Timestamps
    previewed_at: datetime
    confirmed_at: Optional[datetime] = None

    # User corrections (if any)
    user_corrections: Optional[Dict[str, Any]] = None

    class Config:
        json_schema_extra = {
            "example": {
                "document_code": "dip",
                "file_name": "mi_dip.pdf",
                "file_size": 1024000,
                "mime_type": "application/pdf",
                "content_b64": "...",
                "extraction": {"numero_documento": "A12345678"},
                "confidence": 0.95,
                "processor": "gemini",
                "extraction_status": "success",
                "previewed_at": "2026-02-05T10:00:00Z"
            }
        }


class WizardSessionData(BaseModel):
    """
    Complete wizard session data stored in Redis cache.

    This contains ALL wizard state until payment is initiated.
    No database record exists until persist_to_db() is called.
    """
    # Session identification
    session_id: str
    user_id: str

    # Workflow info
    workflow_code: str
    solicitud_type: str  # expedicion, renovacion, duplicado
    sub_type: Optional[str] = None  # NUEVO, PERDIDA, ROBO, DETERIORO, etc.
    motivo: Optional[str] = None  # For renovacion: VENCIMIENTO, PERDIDA, etc.

    # Entity assignment
    entity_code: Optional[str] = None

    # Session status
    status: WizardSessionStatus = WizardSessionStatus.ACTIVE

    # Documents (by document_code)
    documents: Dict[str, SessionDocumentData] = Field(default_factory=dict)

    # Merged extracted data (from all documents)
    extracted_data: Dict[str, Any] = Field(default_factory=dict)

    # User-edited form data
    form_data: Dict[str, Any] = Field(default_factory=dict)

    # Minor info (for pasaporte)
    is_minor: bool = False

    # Current step in wizard
    current_step: int = 1
    current_step_id: Optional[str] = None

    # Tariff calculation (computed on prepare_for_payment)
    tariff: Optional[Dict[str, Any]] = None

    # Appointment selection (cached before payment, not a real hold)
    appointment_data: Optional[Dict[str, Any]] = None

    # Site selection (for non-appointment workflows)
    site_selection: Optional[Dict[str, Any]] = None

    # Validation results
    validation_results: Optional[List[Dict[str, Any]]] = None
    has_errors: bool = False

    # Timestamps
    created_at: datetime
    updated_at: datetime
    expires_at: datetime

    # Audit
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None

    class Config:
        use_enum_values = True


# =============================================================================
# REQUEST/RESPONSE MODELS
# =============================================================================

class WizardSessionCreate(BaseModel):
    """Request to start a new wizard session."""
    workflow_code: str = Field(..., min_length=1, max_length=100)
    solicitud_type: str = Field(default="expedicion", description="expedicion, renovacion, duplicado")
    sub_type: Optional[str] = Field(None, description="Workflow-specific sub-type")
    motivo: Optional[str] = Field(None, description="Reason for renovacion")
    is_minor: bool = Field(default=False, description="Is the applicant a minor?")

    class Config:
        json_schema_extra = {
            "example": {
                "workflow_code": "PASAPORTE",
                "solicitud_type": "renovacion",
                "motivo": "VENCIMIENTO",
                "is_minor": False
            }
        }


class WizardSessionResponse(BaseModel):
    """Response after starting or retrieving a wizard session."""
    session_id: str
    workflow_code: str
    solicitud_type: str
    sub_type: Optional[str] = None
    motivo: Optional[str] = None
    is_minor: bool = False

    status: WizardSessionStatus
    current_step: int
    current_step_id: Optional[str] = None

    # Documents summary (without content_b64)
    documents_count: int = 0
    documents_uploaded: List[str] = Field(default_factory=list)

    # Form data (for restore)
    form_data: Dict[str, Any] = Field(default_factory=dict)
    extracted_data: Dict[str, Any] = Field(default_factory=dict)

    # Tariff (if calculated)
    tariff: Optional[Dict[str, Any]] = None

    # Session timing
    created_at: datetime
    updated_at: datetime
    expires_at: datetime
    ttl_seconds: int

    # Required documents for this workflow/solicitud_type
    required_documents: List[Dict[str, Any]] = Field(default_factory=list)

    # Workflow capabilities
    requires_appointment: bool = False
    entity_code: Optional[str] = None

    # Appointment selection (cached in session, not a real hold)
    appointment_data: Optional[Dict[str, Any]] = None

    # Site selection (for non-appointment workflows)
    site_selection: Optional[Dict[str, Any]] = None

    class Config:
        use_enum_values = True
        json_schema_extra = {
            "example": {
                "session_id": "wizard_session_abc123",
                "workflow_code": "PASAPORTE",
                "solicitud_type": "renovacion",
                "status": "active",
                "current_step": 3,
                "documents_count": 2,
                "documents_uploaded": ["dip", "pasaporte_antiguo"],
                "expires_at": "2026-02-05T10:30:00Z",
                "ttl_seconds": 1800
            }
        }


class WizardDocumentPreviewResponse(BaseModel):
    """Response after previewing a document extraction."""
    session_id: str
    document_code: str
    document_name: Optional[str] = None
    file_name: str
    file_size: int

    # Extraction results
    extraction: Dict[str, Any]
    confidence: float
    processor: str
    extraction_status: str
    needs_correction: bool = False

    # Risk analysis
    risk_analysis: Optional[Dict[str, Any]] = None

    # Cross-document validation (if applicable)
    cross_validation: Optional[Dict[str, Any]] = None

    # Session timing
    expires_at: datetime
    ttl_seconds: int

    class Config:
        json_schema_extra = {
            "example": {
                "session_id": "wizard_session_abc123",
                "document_code": "dip",
                "file_name": "mi_dip.pdf",
                "file_size": 1024000,
                "extraction": {"numero_documento": "A12345678"},
                "confidence": 0.95,
                "processor": "gemini",
                "extraction_status": "success",
                "needs_correction": False,
                "expires_at": "2026-02-05T10:30:00Z",
                "ttl_seconds": 1800
            }
        }


class WizardDocumentConfirmRequest(BaseModel):
    """Request to confirm document extraction (with optional corrections)."""
    document_code: str
    confirmed_data: Dict[str, Any] = Field(..., description="User-confirmed or corrected data")
    user_notes: Optional[str] = Field(None, max_length=500)


class WizardFormDataSaveRequest(BaseModel):
    """Request to save form data to session."""
    form_data: Dict[str, Any] = Field(..., description="Form data to save")
    step_id: Optional[str] = Field(None, description="Current step ID")


class PaymentMethodInfoResponse(BaseModel):
    """Payment method info for frontend display."""
    code: str
    label_es: str
    label_en: str
    label_fr: str
    processor_type: str
    requires_phone: bool
    requires_redirect: bool
    requires_agent_validation: bool


class WizardPreparePaymentResponse(BaseModel):
    """Response after preparing for payment."""
    session_id: str
    ready_for_payment: bool

    # Tariff breakdown
    tariff: Dict[str, Any]
    total_amount: float
    currency: str = "XAF"

    # Validation status
    validation_passed: bool
    errors: List[Dict[str, Any]] = Field(default_factory=list)
    warnings: List[Dict[str, Any]] = Field(default_factory=list)

    # Documents status
    all_documents_uploaded: bool
    missing_documents: List[str] = Field(default_factory=list)

    # Payment methods
    payment_methods: List[PaymentMethodInfoResponse] = Field(default_factory=list)
    default_payment_method: Optional[str] = None

    class Config:
        json_schema_extra = {
            "example": {
                "session_id": "wizard_session_abc123",
                "ready_for_payment": True,
                "tariff": {"base_amount": 50000, "supplements": [], "total_amount": 50000},
                "total_amount": 50000,
                "currency": "XAF",
                "validation_passed": True,
                "errors": [],
                "warnings": [],
                "all_documents_uploaded": True,
                "missing_documents": []
            }
        }


class WizardPersistResult(BaseModel):
    """Result after persisting session to database."""
    success: bool
    service_request_id: Optional[UUID] = None
    reference: Optional[str] = None
    payment_id: Optional[str] = None

    error: Optional[str] = None
    error_code: Optional[str] = None

    class Config:
        json_schema_extra = {
            "example": {
                "success": True,
                "service_request_id": "123e4567-e89b-12d3-a456-426614174000",
                "reference": "PAS-2026-00001",
                "payment_id": "pay_abc123"
            }
        }


class WizardInitiatePaymentRequest(BaseModel):
    """Request to atomically persist session + initiate payment."""
    payment_method: str = Field(..., description="Payment method: mobile_money, card, bank_transfer, cash, check")
    phone_number: Optional[str] = Field(None, description="Phone number (required for mobile_money)")

    class Config:
        json_schema_extra = {
            "example": {
                "payment_method": "mobile_money",
                "phone_number": "+240222123456"
            }
        }


class WizardInitiatePaymentResponse(BaseModel):
    """Response after atomic persist + payment initiation."""
    # Persist result
    success: bool
    service_request_id: Optional[UUID] = None
    reference: Optional[str] = None

    # Payment result
    payment_id: Optional[str] = None
    payment_reference: Optional[str] = None
    payment_status: Optional[str] = None
    redirect_url: Optional[str] = None
    requires_action: bool = False
    action_type: Optional[str] = None  # "redirect", "agent_validation_cash", "agent_validation_check"
    message_es: Optional[str] = None
    expires_at: Optional[datetime] = None

    # Workflow capabilities (for frontend navigation)
    requires_appointment: bool = False

    # Appointment confirmation (set when appointment was held+confirmed atomically)
    appointment_confirmed: bool = False
    appointment_date: Optional[str] = None
    appointment_time: Optional[str] = None
    appointment_location: Optional[str] = None

    # Error info
    error: Optional[str] = None
    error_code: Optional[str] = None

    class Config:
        json_schema_extra = {
            "example": {
                "success": True,
                "service_request_id": "123e4567-e89b-12d3-a456-426614174000",
                "reference": "PAS-2026-00001",
                "payment_id": "pay_abc123",
                "payment_status": "pending",
                "redirect_url": "https://pay.bange.gq/checkout/abc123",
                "requires_action": True,
                "action_type": "redirect",
                "requires_appointment": True,
            }
        }


# =============================================================================
# SITE SELECTION & APPOINTMENT SELECTION REQUEST MODELS
# =============================================================================

class SiteSelectionRequest(BaseModel):
    """Request to save site selection (non-appointment workflows)."""
    entity_location_id: UUID = Field(..., description="FK to entity_locations table")
    location_name: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    entity_code: Optional[str] = Field(None, max_length=50)

    class Config:
        json_schema_extra = {
            "example": {
                "entity_location_id": "550e8400-e29b-41d4-a716-446655440000",
                "location_name": "CNEDOGE Malabo",
                "city": "Malabo",
            }
        }


class AppointmentSelectionRequest(BaseModel):
    """Request to save appointment selection in session cache."""
    entity_location_id: UUID = Field(..., description="FK to entity_locations table")
    location_name: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    appointment_date: date = Field(..., description="Appointment date (YYYY-MM-DD)")
    appointment_time: time = Field(..., description="Appointment time (HH:MM:SS)")
    slot_config_id: Optional[UUID] = Field(None, description="FK to appointment_slot_configs")

    class Config:
        json_schema_extra = {
            "example": {
                "entity_location_id": "550e8400-e29b-41d4-a716-446655440000",
                "location_name": "CNEDOGE Malabo",
                "city": "Malabo",
                "appointment_date": "2026-03-15",
                "appointment_time": "09:00:00",
            }
        }
